File: algorithm/프로그래머스/해시/완주하지못한선수.py
def solution(participant, completion):

    hash_table = {} # 파이썬엔 딕셔너리가 있으니까 이걸 해시테이블로 사용

    for i in participant: # 참가자들을 테이블에 넣기
        try:
            if hash_table[i]: # 테이블에 있으면 동명이인 1씩 추가
                hash_table[i] +=1
        except:
            hash_table[i] = 1 #테이블에 존재하지 않으면 추가
            
    for j in completion: #완주자들 확인하고 1씩 빼주기
        hash_table[j] -= 1
    
    for k in participant: #테이블에 남아있는 이름 확인 후 반환
        if hash_table[k] == 1:
            return k

File: algorithm/C++, Java/완주하지못한선수.java
import java.util.HashMap;

class Solution{
    public String solution(String[] participant, Stirng[ completion]){

        String answer = "";
        HashMap<Stirng, Integer> hm = new HashMap<>();
        // getOrDefault: 찾는 키가 존재한다면 키의 값을 반환, 없으면 기본값을 반환
        for (String player : participant) hm.put(player, hm.getOrDefault(player, 0)+1);
        for (Stirng player : completion) hm.put(player, hm.get(player)-1);

        for (String key : hm.keySet()){
            if (hm.get(key) != 0){
                answer = key;
            }
        }
        return answer;
    }
}